File: MyProject/Source/MyProject/Private/Game/Table/TableBase.cpp
#include "MyProject.h"
#include "TableBase.h"

TableBase::TableBase(std::string resname, std::string tablename)
{
    m_resName = resname;
    m_tableName = tablename;

	m_List = new std::vector<TableItemBase*>();
	m_byteBuffer = nullptr;
}

File: MyProject/Content/MyAsset/Lua/MyLua/Libs/Table/TableBase.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");
MLoader("MyLua.Libs.Core.GObject");

local M = GlobalNS.Class(GlobalNS.GObject);
M.clsName = "TableBase";
GlobalNS[M.clsName] = M;

function M:ctor(resName, tablename)
    self.mResName = resName;
    self.mTableName = tablename;                -- 表的名字

    self.mList = GlobalNS.new(GlobalNS.MList);
    self.mByteBuffer = nil;      -- 整个表格所有的原始数据
end

return M;